obspy/imaging/spectrogram.py

```diff
@@ -151,16 +165,23 @@ def spectrogram(data, samp_rate, per_lap=0.9, wlen=None, log=False,
     vmax = specgram.min() + vmax * _range
     norm = Normalize(vmin, vmax, clip=True)
 
-    if not axes:
+    if axes:
+        ax = axes
+        fig = ax.get_figure()
+    else:
         fig = plt.figure()
         ax = fig.add_subplot(111)
-    else:
-        ax = axes
 
     # calculate half bin width
     halfbin_time = (time[1] - time[0]) / 2.0
     halfbin_freq = (freq[1] - freq[0]) / 2.0
 
+    if starttime is not None:
+        time = date2num([starttime + (t + halfbin_time) for t in time])
```

Comment:
Thanks!
And.. doh, I think we should really add a method to UTCDateTime, something like `UTCDateTime.to_mpl_float()` or something..